File: 55-jump-game/55-jump-game.java
class Solution {
   //oolean res = false;
    public boolean canJump(int[] nums) {
        if(nums.length==1){
            return true;
        }
        int[] dp = new int[nums.length];
        dp[0] = nums[0];
        for(int i=0;i<nums.length;i++){
            for(int j=i+1;j<=dp[i];j++){
                dp[j] = j + nums[j];
                if(dp[j]>=nums.length-1){
                    return true;
                }
            }
        }
        return false;
    }
}

File: 55-jump-game/55-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        start,destination = len(nums)-2,len(nums)-1 # this is greedy solution; here we take two pointers, and iterate from the back of the array
        # at each start pointer, we check if that can be reached to the destination pointer or not; if so, then this start pointer becomes the new destination and the element before it becomes the new start element; check the neetcode solution for it
        while start>=0:
            if start+nums[start]>=destination:
                destination = start
            start -= 1
        return destination==0
                
            